src/install-pnpm/run.ts

```diff
@@ -2,14 +2,28 @@ import { addPath, exportVariable } from '@actions/core'
 import { spawn } from 'child_process'
 import { execPath } from 'process'
 import path from 'path'
-import { remove, ensureFile, writeFile } from 'fs-extra'
+import { remove, ensureFile, writeFile, readFile } from 'fs-extra'
 import fetch from '@pnpm/fetch'
 import { Inputs } from '../inputs'
 
 export async function runSelfInstaller(inputs: Inputs): Promise<number> {
   const { version, dest } = inputs
-  const target = version ? `pnpm@${version}` : 'pnpm'
   const pkgJson = path.join(dest, 'package.json')
+  let target: string
+
+  if (!version) {
+    const packageManager = JSON.parse(await readFile(pkgJson, 'utf8')).packageManager
+    if (packageManager) {
```

Comment:
If you resolve https://github.com/pnpm/action-setup/pull/33/files#r811571211 with a `typeof`, then this line becomes unnecessary.